api/Port.go

```diff
@@ -276,3 +285,118 @@ func (vrsConnection VRSConnection) processUpdates(updates *libovsdb.TableUpdates
 	}
 	return nil
 }
+
+// AddPortToAlubr0 adds Nuage port to alubr0 bridge
+func AddPortToAlubr0(ovs *libovsdb.OvsdbClient, intfName string, entityName string, entityUUID string) error {
+
+	namedPortUUID := "port"
+	namedIntfUUID := "intf"
+	var err error
+
+	// 1) Insert a row for Nuage port in OVSDB Interface table
+	extIDMap := make(map[string]string)
+	intfOp := libovsdb.Operation{}
+	intf := make(map[string]interface{})
+	intf["name"] = intfName
+	extIDMap["vm-name"] = entityName
+	extIDMap["vm-uuid"] = entityUUID
+	intf["external_ids"], err = libovsdb.NewOvsMap(extIDMap)
+	if err != nil {
+		return err
+	}
+	// interface table ops
+	intfOp = libovsdb.Operation{
+		Op:       "insert",
+		Table:    interfaceTable,
+		Row:      intf,
+		UUIDName: "intf",
+	}
+
+	// 2) Insert a row for Nuage port in OVSDB Port table
+	portOp := libovsdb.Operation{}
+	port := make(map[string]interface{})
+	port["name"] = intfName
+	port["interfaces"] = libovsdb.UUID{namedIntfUUID}
+	port["external_ids"], err = libovsdb.NewOvsMap(extIDMap)
+	if err != nil {
+		return err
+	}
+	portOp = libovsdb.Operation{
+		Op:       "insert",
+		Table:    portTable,
+		Row:      port,
+		UUIDName: "port",
+	}
+
+	// 3) Mutate the Ports column of the row in the Bridge table with new Nuage port
+	mutateUUID := []libovsdb.UUID{libovsdb.UUID{namedPortUUID}}
+	mutateSet, _ := libovsdb.NewOvsSet(mutateUUID)
+	mutation := libovsdb.NewMutation("ports", "insert", mutateSet)
+	condition := libovsdb.NewCondition("name", "==", bridgeName)
+	mutateOp := libovsdb.Operation{
+		Op:        "mutate",
+		Table:     bridgeTable,
+		Mutations: []interface{}{mutation},
+		Where:     []interface{}{condition},
+	}
+
+	operations := []libovsdb.Operation{intfOp, portOp, mutateOp}
+	reply, err := ovs.Transact(OvsDBName, operations...)
+	if err != nil || len(reply) < len(operations) {
+		return fmt.Errorf("Problem mutating row in the OVSDB Bridge table for alubr0")
+	}
+
+	return nil
+}
+
+// RemovePortFromAlubr0 will remove a port from alubr0 bridge
+func RemovePortFromAlubr0(ovs *libovsdb.OvsdbClient, portName string) error {
```

Comment:
Indeed. I was working on making the same change while you added the comment. Thanks for pointing this out.